app/models/question.rb

```diff
@@ -1,6 +1,17 @@
 class Question < ActiveRecord::Base
 
   has_many :answers
-  belongs_to :user
+  belongs_to :user, inverse_of: :questions
+
+  def self.all_by_query_params(params)
+    questions = self.where(private: false)
+    if params[:question_name]
+      questions = questions.where('title LIKE ?', '%' + params[:question_name] + '%')
+    elsif params[:answer_name]
+      questions = questions.joins(:answers).where('answers.body LIKE ?', '%' + params[:answer_name] + '%')
+    else
+      questions
+    end
+  end
```

Comment:
@prakashsejwani, how would you modify this to handle multiple queries?

I.e., what if I wanted to request `/questions?question_name=hi&answer_name=no`? Right now, only the first query parameter would be used.